### PR TITLE
doc(zlx): Add HER algorithm doc; Polish bitflip env doc

### PR DESCRIPTION
**Pull request by LuciusMos**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ enum_tools
 sphinx-toolbox
 plantumlcli>=0.0.2
 sphinx-autobuild
-git+http://github.com/opendilab/DI-engine@main
+# git+http://github.com/opendilab/DI-engine@main
```
Comment (PaParaZz1) [inline requirements.txt:8]: uncomment it

File: source/hands_on/her.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+   * - environment
+     - best mean reward
+     - evaluation results
+     - config link
+   ..   - comparison
+   * - BitFLip-15bit
+     - 0.9
+     - .. image:: images/benchmark/bitflip15bit-herdqn.png
+     - `config_link <https://github.com/opendilab/DI-engine/blob/main/dizoo/classic_control/bitflip/config/bitflip_her_dqn_config.py>`_
+   ..   - | Tianshou(20)
```
[the remark this answers is not in the document]
Comment (LuciusMos) [reply source/hands_on/her.rst:61]: Comparison with other frameworks is skipped in this algo doc

File: source/hands_on/her.rst [hunk 3]
```diff
@@ -53,17 +53,14 @@ Benchmark
...
      - `config_link <https://github.com/opendilab/DI-engine/blob/main/dizoo/classic_control/bitflip/config/bitflip_her_dqn_config.py>`_
-   ..   - | Tianshou(20)
    * - BitFLip-20bit
      - 0.9
      - .. image:: images/benchmark/bitflip20bit-herdqn.png
      - `config_link <https://github.com/opendilab/DI-engine/blob/main/dizoo/classic_control/bitflip/config/bitflip_her_dqn_config.py>`_
-   ..   - | Tianshou(16245)
 
 
 P.S.
```
Comment (norman26625) [inline source/hands_on/her.rst:66]: i can‘t see any benchmark pics when i view this file，Is this because of previrew on github？

Comment (norman26625) [inline source/hands_on/her.rst:66]: reference has a standard format, if you don't find it, please contact me (hs)

File: source/hands_on/her.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+HER's core is to **learn from failures**. HER proposes that even if a task's final goal is not achieved, the episode is still of use, because other goals might be achieved by the way. So HER creates a new episode from the original episode, labeling the new one with new rewards by changing the desired task goal.
+
+Quick Facts
+-------------
+1. HER is not a seperate algorithm, it can be **combined with any off-policy** algorithm.
+
+2. HER needs to define rewards after changing the desired goal, but it does not need to be the same as the env's reward function.
+
+3. In DI-engine's implementation, HER collects episode data and push into buffer, then samples episodes and modifies the reward, which will be explained in detail in Pseudo-code part.
```
[the remark this answers is not in the document]
Comment (LuciusMos) [reply source/hands_on/her.rst:16]: 请问具体是什么问题呢？

File: source/hands_on/her.rst (hunk 3 above)
Comment (LuciusMos) [reply source/hands_on/her.rst:66]: Yes, you can render it by running ``make html`` at root path of ``DI-engine-docs``

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**